cpp/src/arrow/array-test.cc

```diff
@@ -58,6 +58,29 @@ TEST_F(TestArray, TestLength) {
   auto data = std::make_shared<PoolBuffer>(pool_);
   std::unique_ptr<Int32Array> arr(new Int32Array(100, data));
   ASSERT_EQ(arr->length(), 100);
+
+  // test slice functions 
+  auto mut_data = data;
+  mut_data->Reserve(100);
+  std::unique_ptr<Int32Array> mut_arr(new Int32Array(25, mut_data));
+
+  auto raw_data = std::dynamic_pointer_cast<MutableBuffer>(mut_arr->data())->mutable_data();
+
+  for(uint32_t i = 0; i < 10; i++) raw_data[i] = i;
+
+  auto sliced_arr_1 = mut_arr->slice(1, 9);
+  auto sliced_data_1 = sliced_arr_1->data()->data();
+  ASSERT_EQ(sliced_arr_1->length(), 9);
+  ASSERT_EQ(memcmp(raw_data + sizeof(uint32_t), sliced_data_1, sliced_arr_1->length()), 0);
+
+  auto sliced_arr_2 = mut_arr->slice(-9, -1);
```

Comment:
Eh....
If there are no more side effects for the code.
I will keep it.
Thanks. :)